cont.c

```diff
@@ -3018,6 +3084,41 @@ rb_fiber_scheduler_interface_kernel_sleep(VALUE self)
 {
 }
 
+/*
+ *  Document-method: SchedulerInterface#address_resolve
+ *  call-seq: address_resolve(hostname) -> array_of_stings or nil
+ *
+ *  Invoked by Socket::getaddrinfo and is expected to provide hostname resolution
+ *  in a non-blocking way.
```

Comment:
This hook is also called by a wide variety of other methods in Ruby. Any (non-reverse) DNS query should invoke this hook. Here's a doc proposal:

```
Invoked by any method that performs a non-reverse DNS lookup. The most notable method is Addrinfo::getaddrinfo, but there are many other, like Socket::getaddrinfo, IPSocket.getaddress, TCPServer.new etc.
```

Here's the full list of methods: https://gist.github.com/bruno-/d2cf7690b0aa4bfa03a8f6823102ffc8

Reply:
Thanks, added.